include/ring_buffer.h

```diff
@@ -116,6 +149,45 @@ static inline void sys_ring_buf_init(struct ring_buf *buf, u32_t size,
 
 }
 
+/** @brief Determine if head and tail indicates empty ring buffer.
+ *
+ * @note Function for internal use.
+ *
+ * @param head Ring buffer Head.
+ * @param tail Ring buffer Tail.
+ *             Ring buffer
+ * @return 1 if empty, 0 otherwise.
+ */
+static inline int sys_ring_buf_custom_is_empty(u32_t head, u32_t tail)
+{
+	return (head == tail);
+}
+
+/** @brief Determine free space based on ring buffer parameters.
+ *
+ * @note Function for internal use.
+ *
+ * @param size Ring buffer size.
+ * @param head Ring buffer head.
+ * @param tail Ring buffer tail.
+ *
+ *  @return Ring buffer free space (in 32-bit words or bytes).
+ */
+static inline int sys_ring_buf_custom_space_get(u32_t size, u32_t head,
+						u32_t tail)
+{
+	if (sys_ring_buf_custom_is_empty(head, tail)) {
```

Comment:
Why this special case? Seems not needed.

Reply:
removed, in fact seems not needed.